BlueDBOnDisk/src/main/java/io/bluedb/disk/segment/Segment.java

```diff
@@ -151,6 +128,30 @@ private static boolean inTimeRange(long minTime, long maxTime, BlueKey key) {
 			return key.getGroupingNumber() >= minTime && key.getGroupingNumber() <= maxTime;
 		}
 	}
+	
+	protected static <T extends Serializable> T remove(BlueKey key, List<BlueEntity<T>> entities) {

```

Comment:
Usually when I do something like this I just us an iterator so that I can call .remove on it if it matches.

Since this is an array list it is fine to call get(i) and remove(i), but if you're doing this way I would change the entities parameter to be an ArrayList to ensure that nobody calls this method with a LinkedList or something that would be incredibly inefficient.
I would change the entities parameter to an ArrayList if we're going to call get(i) in the method. I know you are passing in an ArrayList but this enforces